src/components/dataset/query/sidebar/filter/FreetextFilter.jsx

```diff
@@ -21,36 +36,88 @@ export class FreetextFilter extends React.PureComponent {
     handleChange(value);
   };
 
-  handleReturn = event => {
+  handleReturn = (event) => {
     const { handleFilters } = this.props;
     if (event.key === 'Enter') {
       handleFilters();
     }
   };
 
+  onChange = (event) => {
+    const { handleChange } = this.props;
+    const { value } = event.target;
+    let selections;
+    if (value.includes(',')) {
+      selections = value.split(',');
+    } else if (value.includes(' ')) {
+      selections = value.split(' ');
+    } else {
+      return;
+    }
+    const trimmed = _.map(selections, _.trim);
+    const nonEmpty = _.filter(trimmed, (t) => t !== '');
+    handleChange(nonEmpty);
```

Comment:
I think this can be simplified a bit:  Using regex should return a trimmed array of strings split at " " and ",". Then you can just check if the length of the array is greater than 1: 
`const selections = value.split(/[ ,]+/)`

Reply:
I like that a lot better. Thanks for the tip!!